src/components/Header/TopNav/LogoLink.js

```diff
@@ -23,52 +18,63 @@ const StyledLink = styled(Link)`
-  const isSpecialityPage = path.includes('speciality')
+const getLogoColors = ({ isSpecialityPage, slug }) => {
+  // This isn't very nice I know...
+  // In reality the specialitiesFills shouldn't be in the code
+  // but in contentful. If the content type doesn't have a fill hex
+  // set in the CMS then just fallback to white.
+  const fillColorInitial = isSpecialityPage
+    ? logoColors.specialitiesFills[slug] || logoColors.specialitiesFills.default
+    : logoColors['default']
 
-  const serviceTitle = isServicePage ? path.match(servicesRegExp)[0] : null
-  const service = isSpecialityPage ? getService(path) : serviceTitle
-  const speciality = isSpecialityPage ? getSpeciality(path) : null
+  const fillColorHover =
+    logoColors[isSpecialityPage ? 'specialityHover' : 'defaultHover']
 
-  let originalFillColor = theme.colors.white
+  const textColor =
+    logoColors[isSpecialityPage ? 'specialityText' : 'defaultText']
 
-  if (isServicePage) {
-    originalFillColor = logoColors['default']
-  }
-  if (isSpecialityPage && logoColors[service][speciality]) {
-    originalFillColor = logoColors[service][speciality]
+  return {
+    fillColorInitial,
+    fillColorHover,
+    textColor
   }
+}
 
-  const [fillColor, setFillColor] = useState(originalFillColor)
+const LogoLink = ({ slug, isServicePage, isSpecialityPage, isHomePage }) => {
+  const renderSvg = isServicePage || isSpecialityPage || isHomePage
+
+  const { fillColorInitial, fillColorHover, textColor } = getLogoColors({
+    isSpecialityPage,
+    slug
+  })
+
+  const [fillColor, setFillColor] = useState(fillColorInitial)
+
+  if (renderSvg) {
+    return isServicePage || isSpecialityPage ? (
+      <StyledLink
+        to="/"
+        title="Return to Homepage"
+        onMouseEnter={() => setFillColor(fillColorHover)}
+        onMouseLeave={() => setFillColor(fillColorInitial)}
+      >
+        <ServiceSpecialityLogo fillColor={fillColor} textColor={textColor} />
+      </StyledLink>
+    ) : (
+      <HomepageLogo src={HomepageSvg} alt="yld logo" />
+    )
+  }
 
   return (
-    <Fragment>
-      {isSpecialityPage || isServicePage ? (
-        <StyledLink
-          to="/"
-          title="Return to Homepage"
-          onMouseEnter={() =>
-            setFillColor(
-              logoColors[isServicePage ? 'defaultHover' : 'specialityHover']
-            )
-          }
-          onMouseLeave={() => setFillColor(originalFillColor)}
-        >
-          <ServiceSpecialityLogo
-            fillColor={fillColor}
-            textColor={
-              logoColors[isServicePage ? 'defaultText' : 'specialityText']
-            }
-          />
-        </StyledLink>
-      ) : (
-        <Link to="/" title="Return to Homepage">
-          <img role="link" height="48" src={logo} alt="yld logo" />
-        </Link>
-      )}
-    </Fragment>
+    <Link to="/" title="Return to Homepage">
+      <img role="link" height="48" src={logo} alt="yld logo" />
```

Comment:
I'm wondering how this is read by screen-readers. Mostly wondering if the alt should be yld logo - navigate to home, because if it's read as an image rather than link then that cuts navigation out.